apache/superset
# feat(rightmenu): Add Datasets to + Menu and Hide Databases when one has been connected

### SUMMARY
We are adding some extra logic to RightMenu so it checks for what databases the user has connected before displaying options in the dropdown.

```
Plus menu logic:

Connect database: shows for users who can create databases, until their first non examples db is connected
Create dataset: shows for users who can create a dataset as soon as at least one non examples db is connected
[no change] Connect Google Sheet: shows for users who can create databases, always
[no change] Upload CSV, columnar, Excel: shows for users who can create databases all the time (tooltip when no csv uploads enabled, active when csv uploads enabled). For users who can upload but can't connect db, only shows once uploads are enabled on at least one db.
```

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
![error](https://user-images.githubusercontent.com/38889534/191266353-16748a9c-fc3c-482a-8fb3-e4adc4ebe323.gif)

After:
![test](https://user-images.githubusercontent.com/38889534/191266421-fd8b2822-efa6-4fe7-8ebf-3a4d9977bfb5.gif)
![test](https://user-images.githubusercontent.com/38889534/191268853-8e3d7ace-13e8-4d25-912b-862d5e410862.gif)


### TESTING INSTRUCTIONS
1. Login into the app
2. Connect a database that is not example and check the logic form above
3. Remove all connected databases and check the logic from above
4. Clicking on the connect database menu should open the Connect Database modal
5. Clicking on the create dataset menu should open the Create Dataset modal

### ADDITIONAL INFORMATION


- [X] Has associated issue: https://github.com/apache/superset/issues/21569
